pandas/tests/test_style.py

```diff
@@ -0,0 +1,278 @@
+            (1, 0): ['width: 10em', ' height: 80%', 'background: linear-gradient('\
+                    '90deg,#FFC0CB 50.0%, transparent 0%)'],
+            (2, 0): ['width: 10em', ' height: 80%', 'background: linear-gradient('\
+                    '90deg,#FFC0CB 100.0%, transparent 0%)']
+        }
+        self.assertEqual(result, expected)
+
+        result = df.style.bar(color='red', width=50).ctx
+        expected = {
+            (0, 0): ['width: 10em', ' height: 80%', 'background: linear-gradient('\
+                    '90deg,red 0.0%, transparent 0%)'],
+            (1, 0): ['width: 10em', ' height: 80%', 'background: linear-gradient('\
+                    '90deg,red 25.0%, transparent 0%)'],
+            (2, 0): ['width: 10em', ' height: 80%', 'background: linear-gradient('\
+                    '90deg,red 50.0%, transparent 0%)']
+        }
+        self.assertEqual(result, expected)
+
+        df['C'] = ['a'] * len(df)
+        result = df.style.bar(color='red', width=50).ctx
+        self.assertEqual(result, expected)
+        df['C'] = df['C'].astype('category')
+        result = df.style.bar(color='red', width=50).ctx
+        self.assertEqual(result, expected)
+
+    def test_highlight_null(self, null_color='red'):
+        df = pd.DataFrame({'A': [0, np.nan]})
+        result = df.style.highlight_null().ctx
+        expected = {(0, 0): [''],
+                    (1, 0): ['background-color: red']}
+
+    def test_nonunique_raises(self):
+        df = pd.DataFrame([[1, 2]], columns=['A', 'A'])
+        with tm.assertRaises(ValueError):
+            df.style
+
+        with tm.assertRaises(ValueError):
+            Styler(df)
+
+    def test_caption(self):
+        styler = Styler(self.df, caption='foo')
+        result = styler.render()
+        self.assertTrue(all(['caption' in result, 'foo' in result]))
+
+        # override
+        result = styler.render(caption='bar')
+        self.assertTrue(all(['caption' in result, 'bar' in result,
+                             'foo' not in result]))
+
+    def test_uuid(self):
+        styler = Styler(self.df, uuid='abc123')
+        result = styler.render()
+        self.assertTrue('abc123' in result)
+
+        result = styler.render(uuid='123abc')
+        self.assertTrue('123abc' in result)
+
+    def test_table_styles(self):
+        style = [{'selector': 'th', 'props': [('foo', 'bar')]}]
+        styler = Styler(self.df, table_styles=style)
+        result = ' '.join(styler.render().split())
+        self.assertTrue('th { foo: bar; }' in result)
+
+    def test_precision(self):
+        with pd.option_context('display.precision', 10):
+            s = Styler(self.df)
+        self.assertEqual(s.precision, 10)
+        s = Styler(self.df, precision=2)
+        self.assertEqual(s.precision, 2)
+
+        s2 = s.set_precision(4)
+        self.assertTrue(s is s2)
+        self.assertEqual(s.precision, 4)
+
+def test_no_jinja():
```

Comment:
I would put this near the top
you only need to check if you don't have jinja2 which should raise an import error